### PR TITLE
metrics: add some metrics for snapshot-v2 

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What is changed and how it works?

Issue Number: ref #12842

What's Changed:

```commit-message
1. add snapshot size metrics 
2. add snapshot send/recv count metrics 
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
image

Side effects

### Release note 

```release-note

None
```

### CODE REVIEW

File: components/raftstore/src/store/async_io/read.rs [hunk 1]
```diff
@@ -225,13 +226,15 @@ where
...
+                let total_size = tablet.get_engine_used_size().unwrap_or(0);
                 if let Err(e) = self.generate_snap(&snap_key, tablet) {
                     error!("failed to create checkpointer"; "region_id" => region_id, "error" => %e);
                     SNAP_COUNTER.generate.fail.inc();
                 } else {
                     let elapsed = start.saturating_elapsed_secs();
                     SNAP_COUNTER.generate.success.inc();
                     SNAP_HISTOGRAM.generate.observe(elapsed);
+                    SNAPSHOT_SIZE_HISTOGRAM.observe(total_size as f64);
                     info!("snapshot generated"; "region_id" => region_id, "elapsed" => elapsed, "key" => ?snap_key, "for_balance" => for_balance);
```
Comment (BusyJay) [inline components/raftstore/src/store/async_io/read.rs:238]: Should also log it.

File: src/server/tablet_snap.rs [hunk 2]
```diff
@@ -376,6 +379,9 @@ impl<R: RaftExtension> Runnable for TabletRunner<R> {
         match task {
             Task::Recv { stream, sink } => {
                 let task_num = self.recving_count.load(Ordering::SeqCst);
+                STORE_SNAPSHOT_TRAFFIC_GAUGE_VEC
+                    .with_label_values(&["sending"])
```
Comment (nolouch) [inline src/server/tablet_snap.rs:383]: recving?

Comment (bufferflies) [reply src/server/tablet_snap.rs:383]: fixed

File: components/raftstore/src/store/async_io/read.rs (hunk 1 above)
Comment (bufferflies) [reply components/raftstore/src/store/async_io/read.rs:238]: fixed

File: components/raftstore/src/store/async_io/read.rs [hunk 3]
```diff
@@ -225,14 +226,25 @@ where
...
-                    info!("snapshot generated"; "region_id" => region_id, "elapsed" => elapsed, "key" => ?snap_key, "for_balance" => for_balance);
+                    SNAPSHOT_SIZE_HISTOGRAM.observe(total_size as f64);
+                    SNAPSHOT_KV_COUNT_HISTOGRAM.observe(total_keys as f64);
+                    info!(
+                        "snapshot generated";
+                        "region_id" => region_id,
+                        "elapsed" => elapsed,
+                        "key" => ?snap_key,
+                        "for_balance" => for_balance,
+                        "total_size" => total_size
```
Comment (nolouch) [inline components/raftstore/src/store/async_io/read.rs:246]: no need `total_keys`?

File: src/server/tablet_snap.rs [hunk 4]
```diff
@@ -261,12 +267,12 @@ async fn recv_snap_files(
             };
             f.write_all(&chunk[..])?;
             limit.consume(chunk.len()).await;
+            SNAP_LIMIT_TRANSPORT_BYTES_COUNTER_STATIC
```
Comment (nolouch) [inline src/server/tablet_snap.rs:270]: why move here? need comments.  and size is useless?

**Review by nolouch — COMMENTED** [review]
rest lgtm

File: components/raftstore/src/store/async_io/read.rs (hunk 3 above)
Comment (bufferflies) [reply components/raftstore/src/store/async_io/read.rs]: fixed

File: src/server/tablet_snap.rs (hunk 4 above)
Comment (bufferflies) [reply src/server/tablet_snap.rs]: fixed

**Review by tonyxuqqi — APPROVED** [review]

File: src/server/tablet_snap.rs [hunk 5]
```diff
@@ -378,6 +384,9 @@ impl<R: RaftExtension> Runnable for TabletRunner<R> {
         match task {
             Task::Recv { stream, sink } => {
                 let task_num = self.recving_count.load(Ordering::SeqCst);
+                STORE_SNAPSHOT_TRAFFIC_GAUGE_VEC
+                    .with_label_values(&["receiving"])
+                    .set(task_num as i64);
```
Comment (BusyJay) [inline src/server/tablet_snap.rs:389]: The metrics will never go back to 0 again.

Comment (bufferflies) [reply src/server/tablet_snap.rs:389]: good catch

Comment (bufferflies) [reply src/server/tablet_snap.rs]: I have move it to the store heartbeat and impl snapshot stats in next pr.

File: src/server/tablet_snap.rs [hunk 6]
```diff
@@ -408,8 +411,8 @@ impl<R: RaftExtension> Runnable for TabletRunner<R> {
             }
             Task::Send { addr, msg, cb } => {
                 let region_id = msg.get_region_id();
-                if self.sending_count.load(Ordering::SeqCst) >= self.cfg.concurrent_send_snap_limit
-                {
+                let sending_count = self.sending_count.load(Ordering::SeqCst);
```
Comment (BusyJay) [inline src/server/tablet_snap.rs:414]: Why break it into two lines?

Comment (bufferflies) [reply src/server/tablet_snap.rs]: fixed

**Pull request merged**